src/MongoDB.Driver.Core/Core/Misc/SemanticVersion.cs

```diff
@@ -313,5 +321,159 @@ public static bool TryParse(string value, out SemanticVersion result)
         {
             return !(b < a);
         }
+
+        // private methods
+        private bool IsServerVersion(string preRelease)
+        {
+            if (preRelease == null)
+            {
+                return false;
+            }
+
+            var pattern = @"^((?<releaseCandidate>rc\d+)?-?(?<internalBuild>\d+-g[0-9a-f]{4,40})?)$";
+            var match = Regex.Match(preRelease, pattern);
+
+            return match.Groups["releaseCandidate"].Success || match.Groups["internalBuild"].Success;
+        }
+
+        // nested types
+        internal class ServerVersion
+        {
```

Comment:
Static members go at the top of the class:

    #region static
    // public static methods
    // move FromSemanticVersion here
    #endregion

Reply:
Static method `FromSemanticVersion` substituted with `Parse` and `TryParse`. The were not enclosed by the region in `SemanticVersion`, so I did the same for `ServerVersion`. Please let me know if they still need to be moved.
Not done.

Reply:
While it is true that in general we want static members at the top of the class in this case it is more important that the two files SemanticVersion.cs and ServerVersion.cs have the same ordering to facilitate diffs between them.

Reply:
Also this comment is obsolete... it refers to a method that no longer exists.